ui/src/components/SampleView/GridForm.jsx

```diff
@@ -1,42 +1,63 @@
-/* eslint-disable react/jsx-key */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-/* eslint-disable react/jsx-handler-names */
-import './SampleView.css';
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { Row, Col, Form, Button, Table } from 'react-bootstrap';
 import Draggable from 'react-draggable';
 
-export default class GridForm extends React.Component {
-  constructor(props) {
-    super(props);
-    this.use_advanced_settings = false;
-  }
+import './SampleView.css';
+
+function handleContextMenu(e) {
+  e.stopPropagation();
+}
+
+export default function GridForm(props) {
+  const {
+    getGridOverlayOpacity,
+    gridList,
+    gridResultType,
+    removeGrid,
+    rotateTo,
+    saveGrid,
+    selectedGrids,
+    selectGrid,
+    setGridOverlayOpacity,
+    setGridResultType,
+    show,
+    toggleVisibility,
+  } = props;
+
+  const draggableRef = useRef(null);
+  const [position, setPosition] = useState({ x: 20, y: 64 });
 
-  getGridControls() {
+  useEffect(() => {
+    const draggableElement = draggableRef.current;
+    if (draggableElement) {
+      draggableElement.addEventListener('contextmenu', handleContextMenu);
+    }
+
+    return () => {
+      if (draggableElement) {
+        draggableElement.removeEventListener('contextmenu', handleContextMenu);
+      }
+    };
+  }, [show]);
```

Comment:
Did you try simply passing `onContextMenu={evt => evt.stopPropagation()}` to the `Row` element?

https://stackoverflow.com/a/56328162/758806

Reply:
Yes I actually did, but it didn't change the behavior because the parent element (in `SampleImage`) doesn't use onContextMenu either, but addEventlistener, but now that you mention it, maybe changing the behaviour of the parent element here would have been a better solution

Reply:
Upon this comment I tried changing the event call from the parent element from addEventListener('contextmenu') to onContextMenu, however this changed the offset position of where the event happened, which is relatively crucial for the behavior of the event. I am however not too sure about why this change happens, but it seems that changing the parent event call is not an option in this case

Reply:
Hmm, maybe so `stopPropagation` prevents the event from triggering an event handler on a parent element. Maybe you could try with `stopImmediatePropagation` as well to prevent triggering handlers on the same element (I think React actually registers the event handlers on the root element of the application):

```ts
onContextMenu={evt => {
  evt.stopImmediatePropagation();
  evt.stopPropagation();
}
```

(If it works then try with `stopImmediatePropagation` alone.)

Reply:
Tried that as well earlier, unfortunately this never prevented the unwanted behavior. From my understanding the problem is that React's synthetic event system does not prevent calls to events called on the native event systems. The event call of the parent element looks like this
```
    const imageOverlay = document.querySelector('#insideWrapper');
    imageOverlay.addEventListener('contextmenu', this.rightClick, false);
```
When I tried to change it to use `onContextMenu` (i.e. `onContextMenu={this.rightClick}`) instead, it actually worked with your proposed solution, but unfortunately that added another issue of the canvas context menu being misplaced (always on the top left corner of the canvas element).